conjure-go-client/httpclient/internal/random_scorer.go

```diff
@@ -15,23 +15,33 @@
 package internal
 
 import (
+	"math/rand"
 	"net/http"
 )
 
-type noopScorer struct {
+type randomScorer struct {
 	uris []string
+	rand *rand.Rand
 }
 
-func (n *noopScorer) GetURIsInOrderOfIncreasingScore() []string {
-	return n.uris
+func (n *randomScorer) GetURIsInOrderOfIncreasingScore() []string {
+	uris := make([]string, len(n.uris))
+	copy(uris, n.uris)
+	n.rand.Shuffle(len(uris), func(i, j int) {
```

Comment:
What is the reasoning for making this a random scorer instead of the noop scorer?

Reply:
My intent was to keep default behavior as close to current behavior as possible, which retries the URIs in random order.

`TestFailOverDistribution` in `failover_test.go` fails for example because it assumes URIs are randomized.